api-server-go/v1/services/member_service.go

```diff
@@ -0,0 +1,199 @@
+		UpdatedAt:   member.UpdatedAt.Format(time.RFC3339),
+	}
+
+	return response, nil
+}
+
+// UpdateMember updates an existing Member
+func (s *MemberService) UpdateMember(memberID string, req *models.UpdateMemberRequest) (*models.MemberResponse, error) {
+	var member models.Member
+	err := s.db.First(&member, "member_id = ?", memberID).Error
+	if err != nil {
+		return nil, fmt.Errorf("member not found: %w", err)
+	}
+
+	// Check if we need to update the IDP user
+	needsIdpUpdate := false
+	beforeUpdateName := member.Name
+	beforeUpdatePhoneNumber := member.PhoneNumber
+
+	// Update fields if provided
+	if req.Name != nil {
+		member.Name = *req.Name
+		needsIdpUpdate = true
+	}
+	if req.PhoneNumber != nil {
+		member.PhoneNumber = *req.PhoneNumber
+		needsIdpUpdate = true
+	}
+
+	// Update user in IDP if necessary
+	if needsIdpUpdate {
+		ctx := context.Background()
+		userInstance := &idp.User{
+			Email:       member.Email,
+			FirstName:   member.Name,
+			LastName:    "",
+			PhoneNumber: member.PhoneNumber,
+		}
+
+		_, err := (*s.idp).UpdateUser(ctx, member.IdpUserID, userInstance)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update user in IDP: %w", err)
+		}
+
+		slog.Info("Updated user in IDP", "userID", member.IdpUserID)
+	}
+
+	if err := s.db.Save(&member).Error; err != nil {
+		// Rollback IDP user update if DB operation fails (not implemented here)
+		_, err := (*s.idp).UpdateUser(context.Background(), member.IdpUserID, &idp.User{
+			Email:       member.Email,
+			FirstName:   beforeUpdateName,
+			LastName:    "",
+			PhoneNumber: beforeUpdatePhoneNumber,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("failed to update member: %w", err)
+	}
+
+	response := &models.MemberResponse{
+		MemberID:    member.MemberID,
+		IdpUserID:   member.IdpUserID,
+		Name:        member.Name,
+		Email:       member.Email,
+		PhoneNumber: member.PhoneNumber,
+		CreatedAt:   member.CreatedAt.Format(time.RFC3339),
+		UpdatedAt:   member.UpdatedAt.Format(time.RFC3339),
+	}
+
+	return response, nil
+}
+
+// GetMember retrieves a Member by ID
+func (s *MemberService) GetMember(memberID string) (*models.MemberResponse, error) {
+	var result struct {
+		models.Member
+	}
+	err := s.db.Table("members").
+		Where("members.member_id = ?", memberID).
+		First(&result).Error
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch member: %w", err)
+	}
+
+	response := &models.MemberResponse{
+		MemberID:    result.Member.MemberID,
+		IdpUserID:   result.Member.IdpUserID,
+		Name:        result.Member.Name,
+		Email:       result.Member.Email,
+		PhoneNumber: result.Member.PhoneNumber,
+		CreatedAt:   result.Member.CreatedAt.Format(time.RFC3339),
+		UpdatedAt:   result.Member.UpdatedAt.Format(time.RFC3339),
+	}
+
+	return response, nil
+}
+
+// GetAllMembers retrieves all members
+func (s *MemberService) GetAllMembers() ([]models.MemberResponse, error) {
+	var results []struct {
+		models.Member
+	}
```

Comment:
```suggestion
var results []models.Member
```
Can't we use this instead of this line?